pkg/secret/monitor.go

```diff
@@ -0,0 +1,137 @@
+package secret
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+// ObjectKey represents the unique identifier for a resource, used to access it in the cache.
+type ObjectKey struct {
+	// Namespace is the namespace in which the resource is located.
+	Namespace string
+	// Name denotes metadata.name of a resource being monitored by informer
+	Name string
+}
+
+// singleItemMonitor monitors a single resource using a SharedInformer.
+type singleItemMonitor struct {
+	key      ObjectKey
+	informer cache.SharedInformer
+	lock     sync.Mutex
+	stopped  bool
+	stopCh   chan struct{}
+}
+
+// NewObjectKey creates a new ObjectKey for the given namespace and name.
+func NewObjectKey(namespace, name string) ObjectKey {
+	return ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}
+}
+
+// newSingleItemMonitor creates a new singleItemMonitor for the given key and informer.
+func newSingleItemMonitor(key ObjectKey, informer cache.SharedInformer) *singleItemMonitor {
+	return &singleItemMonitor{
+		key:      key,
+		informer: informer,
+		stopped:  true,
+		stopCh:   make(chan struct{}),
+	}
+}
+
+// HasSynced returns true if the informer's cache has been successfully synced.
+func (i *singleItemMonitor) HasSynced() bool {
+	return i.informer.HasSynced()
+}
+
+// StartInformer starts and runs the informer util the provided context is canceled,
+// or StopInformer() is called. It will block, so call via goroutine.
```

Comment:
`Start*` usually doesn't block in kube.  `Run` usually does.  This looks easier to use if it doesn't block.  Is there a use-case for blocking?

Reply:
This function is ultimately calling `informer.Run` which is a blocking call, and for that reason it's also blocking. 
If `StartInformer()` name is misleading, should we name it `RunInformer()`?

Reply:
> This function is ultimately calling `informer.Run` which is a blocking call, and for that reason it's also blocking.
> If `StartInformer()` name is misleading, should we name it `RunInformer()`?

How about making it non-blocking?

Reply:
Updated to make it non-blocking